### PR TITLE
Bump MS²Rescore to 4.0.2 and replace mokapot with ristretto

### PR DESCRIPTION
**Pull request by jonasscheid**
Bumps MS²Rescore from 3.1.5 to 4.0.2 (Bioconda: bioconda/bioconda-recipes#69340). MS²Rescore 4 removed Mokapot/Percolator integration and always rescores with its built-in `ristretto` engine, so the `mokapot` rescoring engine is replaced by `ristretto`; Percolator remains the default and runs downstream as before.

- `bin/ms2rescore_cli.py` rewritten for the 4.x config schema (global fragment tolerance, `rescoring` block, `finetune`, `ms2` generator; `ionmob`/`maxquant` removed). The original search-engine score is restored before writing the idXML so Percolator sees Comet's `expect` as main score; ristretto q-value/PEP are written as metavalues. If ristretto cannot train (no confident targets) in percolator mode, features are still written and Percolator rescores downstream.
- `--deeplc_calibration_set_size` is now actually passed to MS²Rescore; `--ms2pip_model_dir` is only passed when set.
- `test_mokapot` profile/test renamed to `test_ristretto`; schema enum, docs and citations updated. The ristretto engine honours `--fdr_level`: with `peptide_level_fdrs` the peptidoform-level q-value/PEP are written as `q-value`/`PEP`; PSM/peptidoform/peptide rollups are kept as `ristretto_*` metavalues.
- MS²PIP model enum extended with `timsTOF2024`, `HCD2019`, `HCD2021`.

Supersedes #466.

### Local verification

- `-profile test,docker` (Percolator): 235 peptides vs 253 on `dev` with 3.1.5 (214 shared). Main score/orientation in the MS²Rescore idXML verified identical to the Comet input.
- `-profile test_ristretto,docker` (peptide-level FDR): 111 unique peptides (929 PSM rows), 108 shared with Percolator's 229. At PSM level both engines agree (ristretto 1468 vs Percolator 1382 PSMs at 1%). The peptide-level gap is the estimator: ristretto uses `(decoys + 1) / targets`, Percolator `decoys / targets` with π0; on a test set with a handful of decoys among the top peptides the +1 is decisive, on real data it is not.
- All 7 nf-test pipeline suites pass with updated snapshots.

### Notes for review

- The `test_search_presets` and `test_sdrf` snapshots now contain **empty identification results** for the Percolator path. Both datasets are tiny/weak (1007 PSMs per sample; sdrf sample has 0 PSMs at 1% FDR before rescoring). MS²PIP features are per-PSM consistent with 3.1.5 (r≈0.98), the ristretto metavalues have no effect on Percolator (verified by stripping them), and on the presets sample Percolator's peptide-level outcome depends on its random seed alone (seeds 1/3: ~130 peptides, seeds 2/4711: 0; PSM-level: 372). This is Percolator CV instability on ~1000 PSMs with ~100 features, not a pipeline bug. Ristretto identifies +286 PSMs on the presets samples, so switching those profiles to `--rescoring_engine ristretto` may be preferable.
- The Docker Hub `biocontainers/ms2rescore:4.0.2` mirror was not yet populated at the time of writing (quay.io has it).

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mhcquant/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mhcquant _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core pipelines lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/rescore/main.nf [hunk 1]
```diff
@@ -34,18 +46,50 @@ workflow RESCORE {
     // Compute features via ms2rescore
     MS2RESCORE(ch_merged_runs)
 
-    if (params.rescoring_engine == 'mokapot') {
-        log.warn("The rescoring engine is set to mokapot. This rescoring engine currently only supports psm-level-fdr via ms2rescore.")
-        if (params.global_fdr) {
-            log.warn("Global FDR is currently not supported by mokapot. The global_fdr parameter will be ignored.")
+    if (params.rescoring_engine == 'ristretto') {
+        if (params.fdr_level == 'protein_level_fdrs') {
```
Comment (jonasscheid) [inline subworkflows/local/rescore/main.nf]: strip protein level fdr from pipeline, allow via nextflow schema only psm or pep level fdr

File: subworkflows/local/rescore/main.nf [hunk 2]
```diff
@@ -34,18 +46,50 @@ workflow RESCORE {
...
-        // Switch comet e-value to mokapot q-value
+        // Switch comet e-value to ristretto q-value (local, per sample/condition group)
         OPENMS_IDSCORESWITCHER(MS2RESCORE.out.idxml)
-        ch_rescored_runs = OPENMS_IDSCORESWITCHER.out.idxml
+        ch_rescored_local = OPENMS_IDSCORESWITCHER.out.idxml
 
-        // Filter by mokapot q-value
-        OPENMS_IDFILTER_Q_VALUE(ch_rescored_runs.map { group_meta, idxml -> [group_meta, idxml, []] })
-        ch_filter_q_value = OPENMS_IDFILTER_Q_VALUE.out.filtered
+        if (params.global_fdr) {
```
Comment (jonasscheid) [inline subworkflows/local/rescore/main.nf]: There should be a separation between ristretto and percolator as rescoring engine here. The goal should be to keep the percolator processes and if ristretto is defined that this is handled differently.

File: subworkflows/local/rescore/main.nf [hunk 3]
```diff
@@ -34,18 +46,50 @@ workflow RESCORE {
...
+            // Group by search_preset for global FDR, exactly like the Percolator branch
+            OPENMS_IDMERGER_GLOBAL(
+                MS2RESCORE.out.idxml.map { group_meta, idxml -> [group_meta + [id: group_meta.search_preset ?: 'global'], idxml] }.groupTuple()
+            )
+            // Second MS²Rescore pass: all features are already present, so only ristretto runs, dataset-wide
+            MS2RESCORE_GLOBAL(OPENMS_IDMERGER_GLOBAL.out.idxml.map { meta, idxml -> [meta, idxml, [], []] })
+            OPENMS_IDSCORESWITCHER_GLOBAL(MS2RESCORE_GLOBAL.out.idxml)
```
Comment (jonasscheid) [inline subworkflows/local/rescore/main.nf]: every global step is very expensive. If there is an alternative that minizes global read write modules, that solution should be preferred

File: subworkflows/local/rescore/main.nf (hunk 1 above)
Comment (jonasscheid) [reply subworkflows/local/rescore/main.nf]: Done in 3ac4361: `protein_level_fdrs` removed from the `--fdr_level` enum (schema, script, docs, CHANGELOG); the warning here is gone.

File: subworkflows/local/rescore/main.nf (hunk 2 above)
Comment (jonasscheid) [reply subworkflows/local/rescore/main.nf]: Restructured in 3ac4361. The engine branch now only covers what differs: Percolator keeps `PSMFEATUREEXTRACTOR` → `PERCOLATORADAPTER` (and `PERCOLATORADAPTER_GLOBAL` on the merged file); ristretto uses the MS²Rescore output directly (and `MS2RESCORE_GLOBAL` on the merged file). Merge, q-value filter, back-filter, TSV export and PSM tables are written once and shared by both engines.

File: subworkflows/local/rescore/main.nf (hunk 3 above)
Comment (jonasscheid) [reply subworkflows/local/rescore/main.nf]: Agreed. `ms2rescore_cli.py` now writes the ristretto q-value as idXML main score itself (Comet e-value kept as `expect` metavalue, same as the former `IDScoreSwitcher -new_score q-value -old_score expect`), so both `IDSCORESWITCHER` processes are gone (3ac4361). The ristretto global chain is now IDMerger → MS2Rescore → IDFilter, i.e. the same number of global read/writes as the Percolator chain. Verified locally: the second pass restores the e-value from the metavalue and trains on the same 97 features; all 8 nf-tests pass.
